### PR TITLE
New package: darktable

### PR DESCRIPTION
**Pull request by peterbud**
Only builds for x64, 32-bit version won't compile and not supported.

### CODE REVIEW

**Comment by Alexpux** [comment]
@peterbud I have error building:
````
C:/repo/mingw-w64-darktable/src/build-x86_64/data/darktable.desktop: error: file contains at least one line ending with a carriage return, while lines should only be separated by a line feed character. First such line is: "[Desktop Entry]"
````

**Comment by Alexpux** [comment]
Previous error fixed in https://github.com/Alexpux/MINGW-packages/commit/101753506b09501131c6f4753edf9d3c429e7fcb

But now other errors.

**Comment by peterbud** [comment]
@Alexpux , what is the error? Even the original `PKGBUILD` builds properly on Appveyor, if you remove the 32-bit build from the matrix, see the log [here](https://ci.appveyor.com/project/peterbud/mingw-packages/build/1.0.118)

**Comment by Alexpux** [comment]
@peterbud see https://gist.github.com/Alexpux/98f4fa42d877a66c48ab5e5696599b9d

**Comment by peterbud** [comment]
I was able to reproduce the problem with the desktop file validation: `mingw-w64-x86_64-desktop-file-utils `is not a dependency, but apparently you have it installed in your environment. Your [fix](https://github.com/Alexpux/MINGW-packages/commit/101753506b09501131c6f4753edf9d3c429e7fcb) is working fine, but ultimately I'll make sure that the desktop file won't be installed as there is no purpose of that file in Windows, I believe.

Your compile error I was not able to reproduce however, I tried everything, but it just compiled properly. Can you give a try with a clean environment? Or maybe somebody else can also try?

**Comment by Alexpux** [comment]
@peterbud clang not used.
See:
````
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --green --bold --progress-dir=/C/repo/mingw-w64-darktable/src/build-x86_64/CMakeFiles --progress-num=$(CMAKE_PROGRESS_153) "Linking C shared library libdarktable.dll"
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && /C/building/msys64/mingw64/bin/cmake.exe -E remove -f CMakeFiles/lib_darktable.dir/objects.a
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && /C/building/msys64/mingw64/bin/ar.exe cr CMakeFiles/lib_darktable.dir/objects.a @CMakeFiles/lib_darktable.dir/objects1.rsp
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && /C/building/msys64/mingw64/bin/gcc.exe  -march=x86-64 -mtune=generic -O2 -pipe -Wall -fno-strict-aliasing -Wno-format -mms-bitfields -Wshadow -Wtype-limits -Wvla -Wold-style-declaration -Wno-error=varargs -Wno-format-truncation -Wframe-larger-than=32768 -Wstack-usage=32768 -Wlarger-than=524288 -std=c99 -fopenmp -march=native -msse2 -g -mfpmath=sse -O3 -DNDEBUG -O3 -ffast-math -fno-finite-math-only -fexpensive-optimizations -pipe -Wl,--as-needed -Wl,--enable-runtime-pseudo-reloc -shared -o libdarktable.dll -Wl,--out-implib,libdarktable.dll.a -Wl,--major-image-version,0,--minor-image-version,0 -Wl,--whole-archive CMakeFiles/lib_darktable.dir/objects.a -Wl,--no-whole-archive external/LuaAutoC/liblautoc.a external/rawspeed/src/librawspeed.a -lm -lgomp /C/building/msys64/mingw64/lib/libglib-2.0.dll.a -lgtk-3 -lgdk-3 -lgdi32 -limm32 -lshell32 -lole32 -lwinmm -ldwmapi -lsetupapi -lcfgmgr32 -lz -lpangowin32-1.0 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl /C/building/msys64/mingw64/lib/libxml2.dll.a /C/building/msys64/mingw64/lib/libsoup-2.4.dll.a /C/building/msys64/mingw64/lib/libgio-2.0.dll.a /C/building/msys64/mingw64/lib/libgobject-2.0.dll.a /C/building/msys64/mingw64/lib/libglib-2.0.dll.a /C/building/msys64/mingw64/lib/libintl.dll.a /C/building/msys64/mingw64/lib/libgphoto2.dll.a /C/building/msys64/mingw64/lib/libgphoto2_port.dll.a /C/building/msys64/mingw64/lib/libIlmImf.dll.a /C/building/msys64/mingw64/lib/libImath.dll.a /C/building/msys64/mingw64/lib/libIex.dll.a /C/building/msys64/mingw64/lib/libIlmThread.dll.a /C/building/msys64/mingw64/lib/libwebp.dll.a /C/building/msys64/mingw64/lib/liblensfun.dll.a /C/building/msys64/mingw64/lib/libgio-2.0.dll.a /C/building/msys64/mingw64/lib/libglib-2.0.dll.a /C/building/msys64/mingw64/lib/libgthread-2.0.dll.a /C/building/msys64/mingw64/lib/libgmodule-2.0.dll.a /C/building/msys64/mingw64/lib/libpangocairo-1.0.dll.a /C/building/msys64/mingw64/lib/librsvg-2.dll.a /C/building/msys64/mingw64/lib/libsqlite3.dll.a /C/building/msys64/mingw64/lib/libcurl.dll.a /C/building/msys64/mingw64/lib/libpng.dll.a /C/building/msys64/mingw64/lib/libz.dll.a /C/building/msys64/mingw64/lib/libjpeg.dll.a /C/building/msys64/mingw64/lib/libtiff.dll.a /C/building/msys64/mingw64/lib/liblcms2.dll.a /C/building/msys64/mingw64/lib/libjson-glib-1.0.dll.a /C/building/msys64/mingw64/lib/libexiv2.dll.a /C/building/msys64/mingw64/lib/libexpat.dll.a /C/building/msys64/mingw64/lib/libiconv.dll.a /C/building/msys64/mingw64/lib/libflickcurl.dll.a /C/building/msys64/mingw64/lib/libopenjp2.dll.a /C/building/msys64/mingw64/lib/libsecret-1.dll.a /C/building/msys64/mingw64/lib/libgio-2.0.dll.a /C/building/msys64/mingw64/lib/libgobject-2.0.dll.a /C/building/msys64/mingw64/lib/libglib-2.0.dll.a /C/building/msys64/mingw64/lib/libintl.dll.a /C/building/msys64/mingw64/lib/libGraphicsMagick.dll.a -lpsapi -lexchndl -LC:/building/msys64/mingw64/lib -llua -lm /C/building/msys64/mingw64/lib/pugixml-1.8/libpugixml.dll.a /C/building/msys64/mingw64/lib/libosmgpsmap-1.0.dll.a /C/building/msys64/mingw64/lib/libgtk-3.dll.a /C/building/msys64/mingw64/lib/libgdk-3.dll.a /C/Windows/System32/gdi32.dll /C/Windows/System32/imm32.dll /C/Windows/System32/shell32.dll /C/Windows/System32/ole32.dll /C/Windows/System32/winmm.dll /C/Windows/System32/dwmapi.dll /C/Windows/System32/setupapi.dll /C/Windows/System32/cfgmgr32.dll /C/building/msys64/mingw64/lib/libz.dll.a /C/building/msys64/mingw64/lib/libpangowin32-1.0.dll.a /C/building/msys64/mingw64/lib/libpangocairo-1.0.dll.a /C/building/msys64/mingw64/lib/libpango-1.0.dll.a /C/building/msys64/mingw64/lib/libatk-1.0.dll.a /C/building/msys64/mingw64/lib/libcairo-gobject.dll.a /C/building/msys64/mingw64/lib/libcairo.dll.a /C/building/msys64/mingw64/lib/libgdk_pixbuf-2.0.dll.a /C/building/msys64/mingw64/lib/libsoup-2.4.dll.a /C/building/msys64/mingw64/lib/libgio-2.0.dll.a /C/building/msys64/mingw64/lib/libgobject-2.0.dll.a /C/building/msys64/mingw64/lib/libglib-2.0.dll.a /C/building/msys64/mingw64/lib/libintl.dll.a /C/building/msys64/mingw64/lib/libz.dll.a -lm -lgomp -lgtk-3 -lgdk-3 -lgdi32 -limm32 -lshell32 -lole32 -lwinmm -ldwmapi -lsetupapi -lcfgmgr32 -lz -lpangowin32-1.0 -lpangocairo-1.0 -lpango-1.0 -latk-1.0 -lcairo-gobject -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lgobject-2.0 -lglib-2.0 -lintl /C/building/msys64/mingw64/lib/libxml2.dll.a /C/building/msys64/mingw64/lib/libgphoto2.dll.a /C/building/msys64/mingw64/lib/libgphoto2_port.dll.a /C/building/msys64/mingw64/lib/libIlmImf.dll.a /C/building/msys64/mingw64/lib/libImath.dll.a /C/building/msys64/mingw64/lib/libIex.dll.a /C/building/msys64/mingw64/lib/libIlmThread.dll.a /C/building/msys64/mingw64/lib/libwebp.dll.a /C/building/msys64/mingw64/lib/liblensfun.dll.a /C/building/msys64/mingw64/lib/libgthread-2.0.dll.a /C/building/msys64/mingw64/lib/libgmodule-2.0.dll.a /C/building/msys64/mingw64/lib/libpangocairo-1.0.dll.a /C/building/msys64/mingw64/lib/librsvg-2.dll.a /C/building/msys64/mingw64/lib/libsqlite3.dll.a /C/building/msys64/mingw64/lib/libcurl.dll.a /C/building/msys64/mingw64/lib/libpng.dll.a /C/building/msys64/mingw64/lib/libz.dll.a /C/building/msys64/mingw64/lib/libtiff.dll.a /C/building/msys64/mingw64/lib/liblcms2.dll.a /C/building/msys64/mingw64/lib/libjson-glib-1.0.dll.a /C/building/msys64/mingw64/lib/libexiv2.dll.a /C/building/msys64/mingw64/lib/libexpat.dll.a /C/building/msys64/mingw64/lib/libiconv.dll.a /C/building/msys64/mingw64/lib/libflickcurl.dll.a /C/building/msys64/mingw64/lib/libopenjp2.dll.a /C/building/msys64/mingw64/lib/libsecret-1.dll.a /C/building/msys64/mingw64/lib/libGraphicsMagick.dll.a -lpsapi -lexchndl -llua /C/building/msys64/mingw64/lib/libosmgpsmap-1.0.dll.a /C/building/msys64/mingw64/lib/libgtk-3.dll.a /C/building/msys64/mingw64/lib/libgdk-3.dll.a /C/Windows/System32/gdi32.dll /C/Windows/System32/imm32.dll /C/Windows/System32/shell32.dll /C/Windows/System32/ole32.dll /C/Windows/System32/winmm.dll /C/Windows/System32/dwmapi.dll /C/Windows/System32/setupapi.dll /C/Windows/System32/cfgmgr32.dll /C/building/msys64/mingw64/lib/libpangowin32-1.0.dll.a /C/building/msys64/mingw64/lib/libpango-1.0.dll.a /C/building/msys64/mingw64/lib/libatk-1.0.dll.a /C/building/msys64/mingw64/lib/libcairo-gobject.dll.a /C/building/msys64/mingw64/lib/libcairo.dll.a /C/building/msys64/mingw64/lib/libgdk_pixbuf-2.0.dll.a /C/building/msys64/mingw64/lib/libjpeg.dll.a /C/building/msys64/mingw64/lib/pugixml-1.8/libpugixml.dll.a -lkernel32 -luser32 -lgdi32 -lwinspool -lshell32 -lole32 -loleaut32 -luuid -lcomdlg32 -ladvapi32 
	@$(CMAKE_COMMAND) -E cmake_echo_color --switch=$(COLOR) --blue --bold "Detaching debug infos for lib_darktable."
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && /C/building/msys64/mingw64/bin/cmake.exe -E make_directory .debug
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && objcopy --only-keep-debug C:/repo/mingw-w64-darktable/src/build-x86_64/src/libdarktable.dll C:/repo/mingw-w64-darktable/src/build-x86_64/src/libdarktable.dll.dbg
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && objcopy --strip-debug C:/repo/mingw-w64-darktable/src/build-x86_64/src/libdarktable.dll
	cd /C/repo/mingw-w64-darktable/src/build-x86_64/src && objcopy --add-gnu-debuglink=libdarktable.dll.dbg C:/repo/mingw-w64-darktable/src/build-x86_64/src/libdarktable.dll
````

File: mingw-w64-darktable/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  if check_option "debug" "n"; then
+    extra_config+=("-DCMAKE_BUILD_TYPE=Release")
+  else
+    extra_config+=("-DCMAKE_BUILD_TYPE=Debug")
+  fi
+
+  MSYS2_ARG_CONV_EXCL="-DCMAKE_INSTALL_PREFIX=" \
+    ${MINGW_PREFIX}/bin/cmake \
+      -G'MSYS Makefiles' \
+      -DCMAKE_INSTALL_PREFIX=${MINGW_PREFIX} \
```
Comment (LebedevRI) [inline mingw-w64-darktable/PKGBUILD]: @peterbud you clearly don't pass `-DBINARY_PACKAGE_BUILD=ON`.

File: mingw-w64-darktable/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  "${MINGW_PACKAGE_PREFIX}-graphicsmagick"
+  "${MINGW_PACKAGE_PREFIX}-openjpeg2"
+  "${MINGW_PACKAGE_PREFIX}-gtk3"
+  "${MINGW_PACKAGE_PREFIX}-pugixml"
+  "${MINGW_PACKAGE_PREFIX}-libexif"
+  "${MINGW_PACKAGE_PREFIX}-osm-gps-map"
+  "${MINGW_PACKAGE_PREFIX}-libgphoto2"
+  "${MINGW_PACKAGE_PREFIX}-flickcurl"
+  "${MINGW_PACKAGE_PREFIX}-drmingw"
+)
```
Comment (LebedevRI) [inline mingw-w64-darktable/PKGBUILD]: zlib is missing

Comment (peterbud) [reply mingw-w64-darktable/PKGBUILD]: it's indirectly already pulled, libxml2, libxslt depends on that I believe.

File: mingw-w64-darktable/PKGBUILD (hunk 1 above)
Comment (peterbud) [reply mingw-w64-darktable/PKGBUILD]: OK, I can add - funny, I have never used that so far.

File: mingw-w64-darktable/PKGBUILD (hunk 2 above)
Comment (LebedevRI) [reply mingw-w64-darktable/PKGBUILD]: And then those stop pulling it, and?
All direct dependencies should be directly pulled, even if they are already pulled indirectly.

**Comment by peterbud** [comment]
This is excellent news! It's no wonder that you cannot build a 32-bit version, as there is only 64-bit version of darktable:
https://github.com/darktable-org/darktable/blob/master/packaging/windows/BUILD.txt#L5
Will you add then the package to the repo?

**Comment by Alexpux** [comment]
@peterbud I'm build it in home but on work it still fail. I will upload package from home and will try to compare build and try to see the issue

**Pull request merged**